CMakeLists.txt

```diff
@@ -44,18 +44,39 @@ option(STABLEHLO_ENABLE_STRICT_BUILD "Build StableHLO with strict warnings and w
 #-------------------------------------------------------------------------------
 # Project setup and globals
 #-------------------------------------------------------------------------------
-set(STABLEHLO_STANDALONE_BUILD OFF)
+set(STABLEHLO_EXTERNAL_PROJECT_BUILD OFF)
 
-if(CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
-  set(STABLEHLO_STANDALONE_BUILD ON)
+if(NOT (CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR) AND NOT MLIR_BINARY_DIR)
+  # Building as part of LLVM via the external project mechanism.
+  set(STABLEHLO_EXTERNAL_PROJECT_BUILD ON)
+else()
+  # Building standalone.
   project(stablehlo LANGUAGES CXX C)
   set(CMAKE_C_STANDARD 11)
   set(CMAKE_CXX_STANDARD 17)
 endif()
 
-if(NOT STABLEHLO_STANDALONE_BUILD AND NOT MLIR_BINARY_DIR)
-  # Building as part of LLVM via the external project mechanism.
-  set(STABLEHLO_EXTERNAL_PROJECT_BUILD ON)
+# Build with ccache if the package is present
+set(LLVM_CCACHE_BUILD OFF CACHE BOOL "Set to ON for a ccache enabled build")
```

Comment:
I wonder if we can rely on this flag in CI? Currently we use a ccache step to set this up I believe. Doesn't need to be addressed now.

Reply:
@kuhar @GleasonK Good point about Ccache! I think we should keep `LLVM_CCACHE_BUILD` in our build for 1:1 compatibility with MLIR-HLO and not do anything further for now. Down the line, when we sunset MLIR-HLO and folks successfully migrate, we can refactor.